### PR TITLE
Bug 2028106: convert vm,vmi actions to dynamic

### PR DESCRIPTION
**Pull request by glekner**
issue: https://issues.redhat.com/browse/CNV-14855

Topology view Paused VM
Screen Shot 2021-11-18 at 13 10 32

--------------------------

VM List Running VM
Screen Shot 2021-11-18 at 13 11 50

--------------------------

VM Details Page Stopped VM
Screen Shot 2021-11-18 at 13 12 52

--------------------------

### CODE REVIEW

**Comment by yaacov** [comment]
a - we need to remove all the deprecated kebabAction code
b - we need to remove the copySSH until we implement https://issues.redhat.com/browse/CNV-9746 in the UI
c - we want the stop/start migrate/cancel pause/start to be one action, so we will not need to do 'check ? start : strop' we will only have one action that change it's behavior/label depending on vm state
d - we need to handle topology dropdown
e - we need to make sure all disable/enable flags are true the design docs

**Comment by yaacov** [comment]
f - we need to unify the actions arguments, so we can loop on all actions and they will accept the same argument

File: frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx [hunk 1]
```diff
@@ -716,6 +270,49 @@ export const VmActionFactory = {
       }),
     accessReview: asAccessReview(kindObj, vm, 'delete'),
   }),
+  // disabled until https://issues.redhat.com/browse/CNV-9746 is implemented
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx:273]: this code should be removed, when https://issues.redhat.com/browse/CNV-9746
we will not use this code

File: frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx [hunk 2]
```diff
@@ -521,14 +71,17 @@ export const VmActionFactory = {
...
+        isVMRunningOrExpectedRunning(vm, vmi) ||
+        isVMIPaused(vmi),
+      label: i18next.t('kubevirt-plugin~Start'),
       cta: () => {
         if (!vmStatusBundle?.status?.isImporting()) {
           startVM(vm);
         } else {
           confirmModal({
-            title: i18next.t('kubevirt-plugin~ Start Virtual Machine'),
+            title: i18next.t('kubevirt-plugin~ Start'),
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx:84]: why do we have a space before Start ?
do we need to play with css ? confirm model does not print the title correclty ?

File: frontend/packages/kubevirt-plugin/src/actions/provider.ts [hunk 3]
```diff
@@ -44,9 +45,11 @@ export const useVmActionsProvider = (vm: VMKind) => {
           start,
           VmActionFactory.Restart(k8sModel, vm, { vmi }),
           pause,
-          migrate,
           VmActionFactory.Clone(k8sModel, vm, { vmi }),
+          migrate,
           VmActionFactory.OpenConsole(k8sModel, vm, { vmi }),
+          // disabled until https://issues.redhat.com/browse/CNV-9746 is implemented
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/actions/provider.ts:51]: it's removed until we have better way to do ssh, when we do, we will have a different action

File: frontend/packages/kubevirt-plugin/src/actions/provider.ts [hunk 4]
```diff
@@ -74,7 +77,7 @@ export const useVmiActionsProvider = (vm: VMKind) => {
           VmActionFactory.OpenConsole(k8sModel, vm, { vmi }),
           CommonActionFactory.ModifyLabels(k8sModel, vm),
           CommonActionFactory.ModifyAnnotations(k8sModel, vm),
-          VmActionFactory.Delete(k8sModel, vm, { vmi }),
+          VmiActionFactory.Delete(k8sModel, vmi),
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/actions/provider.ts:80]: how will we delete a vm that does not have a vmi ?

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/actions/provider.ts:80]: ahh .. ok

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-page-new.tsx [hunk 5]
```diff
@@ -127,18 +127,24 @@ const VMRow: React.FC<RowFunctionArgs<VMRowObjType, VmStatusResourcesValue>> = (
...
 
   if (vm) {
     model = VirtualMachineModel;
+    objReference = referenceFor(vm);
+    context = { [objReference]: vm };
   } else if (vmi) {
     model = VirtualMachineInstanceModel;
+    objReference = referenceFor(vmi);
+    context = { [objReference]: vmi };
   }
```
Comment (metalice) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm-page-new.tsx:148]: consider:
```suggestion
  let model: K8sKind;
  let objReference;
  let context;

  if (vm) {
    model = VirtualMachineModel;
    objReference = referenceFor(vm);
    context = { [objReference]: vm };
  } else if (vmi) {
    model = VirtualMachineInstanceModel;
    objReference = referenceFor(vmi);
    context = { [objReference]: vmi };
  }
 
```

const dataBuilder = (vm: VMKind, vmi: VMIKind) => {
    return {
      model: (vm && VirtualMachineModel) || (vmi && VirtualMachineInstanceModel),
      objReference: referenceFor(vm || vmi),
      context: { [objReference]: vm || vmi },
    };
  };`

same goes to 2 other files where you use similar

**Pull request merged**